pubsub/subscriptions/subscription_test.go

```diff
@@ -263,6 +264,166 @@ func TestPullMsgsConcurrencyControl(t *testing.T) {
+	if err != nil {
+		t.Fatalf("getOrCreateTopic: %v", err)
+	}
+	defer deadLetterSinkTopic.Delete(ctx)
+	defer deadLetterSinkTopic.Stop()
+
+	buf := new(bytes.Buffer)
+	if err := createSubWithDeadLetter(buf, tc.ProjectID, deadLetterSubID, deadLetterSourceID, deadLetterSinkTopic.String()); err != nil {
+		t.Fatalf("failed to create a subscription with dead lettering: %v", err)
+	}
+	sub := client.Subscription(deadLetterSubID)
+	ok, err := sub.Exists(context.Background())
+	if err != nil {
+		t.Fatalf("failed to check if sub exists: %v", err)
+	}
+	if !ok {
+		t.Fatalf("got none; want sub = %q", deadLetterSubID)
+	}
+	defer sub.Delete(ctx)
+
+	if err := updateDeadLetter(buf, tc.ProjectID, deadLetterSubID, deadLetterSinkTopic.String()); err != nil {
+		t.Fatalf("failed to update a subscriptions's dead letter policy: %v", err)
+	}
+
+	cfg, err := sub.Config(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	got := cfg.DeadLetterPolicy
+	want := &pubsub.DeadLetterPolicy{
+		DeadLetterTopic:     deadLetterSinkTopic.String(),
+		MaxDeliveryAttempts: 20,
+	}
+	if !cmp.Equal(got, want) {
+		t.Fatalf("got cfg: %+v; want cfg: %+v", got, want)
+	}
+
+	if err := removeDeadLetterTopic(buf, tc.ProjectID, deadLetterSubID); err != nil {
+		t.Fatalf("got error removing dead letter policy: %v", err)
+	}
+	cfg, err = sub.Config(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	got = cfg.DeadLetterPolicy
+	if got != nil {
+		t.Fatalf("got dead letter policy: %+v, want nil", got)
+	}
+}
+
+func TestPullMsgsDeadLetterDeliveryAttempts(t *testing.T) {
+	client := setup(t)
+	ctx := context.Background()
+	tc := testutil.SystemTest(t)
+	deadLetterSourceID := topicID + "-dead-letter-source"
+	deadLetterSinkID := topicID + "-dead-letter-sink"
+	deadLetterSubID := subID + "-dead-letter"
+
+	topic, err := getOrCreateTopic(ctx, client, deadLetterSourceID)
+	if err != nil {
+		t.Fatalf("getOrCreateTopic: %v", err)
+	}
+	defer topic.Delete(ctx)
+	defer topic.Stop()
+
+	deadLetterTopic, err := getOrCreateTopic(ctx, client, deadLetterSinkID)
+	if err != nil {
+		t.Fatalf("getOrCreateTopic: %v", err)
+	}
+	defer deadLetterTopic.Delete(ctx)
+	defer topic.Stop()
+
+	buf := new(bytes.Buffer)
+	deadLetterSinkName := fmt.Sprintf("projects/%s/topics/%s", tc.ProjectID, deadLetterSinkID)
+	if err = createSubWithDeadLetter(buf, tc.ProjectID, deadLetterSubID, deadLetterSourceID, deadLetterSinkName); err != nil {
+		t.Fatalf("createSubWithDeadLetter: %v", err)
+	}
+	sub := client.Subscription(deadLetterSubID)
+	defer sub.Delete(ctx)
+
+	if err = publishMsgs(ctx, topic, 1); err != nil {
+		t.Fatalf("publishMsgs: %v", err)
+	}
+
+	if err := pullMsgsDeadLetterDeliveryAttempt(buf, tc.ProjectID, deadLetterSubID); err != nil {
+		t.Fatalf("failed to pull messages: %v", err)
+	}
+	got := buf.String()
+	want := "delivery attempts: 1"
+	if !strings.Contains(got, want) {
+		t.Fatalf("pullMsgsDeadLetterDeliveryAttempts got %s, want %s", got, want)
+	}
```

Comment:
If you did `msg.NAck()` in `PullMsgsDeadLetterDeliveryAttempts`, perhaps the test here can be written a little more interestingly to show 1). delivery attempt number increments 2). it is best effort only.